code/crypto/lib/Loot/Crypto/Bip39.hs

```diff
@@ -46,10 +47,10 @@ instance Exception EncodingError
 
 -- | Turn entropy (pure bytes) into a mnemonic.
 entropyToMnemonic :: HasCallStack => ByteString -> [Text]
-entropyToMnemonic ent = runIdentity $ do
+entropyToMnemonic ent = either impureThrow id . runExcept $ do
```

Comment:
`runExcept` is useless, `Either a` is instance of `MonadError a`.
Maybe you find it more understandable and that's on purpose, I don't know :man_shrugging: 
It's possible to avoid dependency on `mtl` and use `Left` instead of `throwError`, but dependency on `mtl` is ok, so I think it doesn't matter.

Reply:
Yes, it’s more of a conceptual choice. I strongly believe that `Either` should be treated as a generic sum type and it is not right to assume that is‘s only (or even primary) use-case is error handling, so one shouldn’t pre-assign any semantics to it and provide a `Monad` instance.